File: src/025/p025.ts
import { fibonacci } from "../utils";

export const solve1 = (): string => {
  let index = 0;
  for (const n of fibonacci()) {
    index++;
    if (n.toString().length >= 1000) {
      break;
    }
  }
  return index.toString();
};

export const solve = (): string => {
  const phi = (1 + Math.sqrt(5)) / 2;
  const n = (999 * Math.log(10) + Math.log(5) / 2) / Math.log(phi);
  return Math.round(n).toString();
};

File: src/025/p025.py
import math
from src.utils.py.series import fibonacci


def solve1() -> int:
    index = 0
    for num in fibonacci():
        index += 1
        if len(str(num)) >= 1000:
            break
    return index


def solve() -> int:
    phi = (1 + math.sqrt(5)) / 2
    return round((999 * math.log(10) + math.log(5) / 2) / math.log(phi))
